prusti-interface/src/environment/polonius_info.rs

```diff
@@ -429,24 +447,41 @@ fn add_fake_facts<'a, 'tcx: 'a>(
-                        last_loan_id += 1;
+                    if let Some(var_region) = place_regions.for_local(arg) {
+                        let loan = new_loan();
+                        debug!("var_region = {:?} loan = {:?}", var_region, loan);
+                        borrow_region.push((var_region, loan, point));
                         argument_moves.push(loan);
                     }
                 }
             } else if is_assignment(&mir, location) {
-                let (_region1, region2) = regions.pop().unwrap();
-                let new_loan = facts::Loan::from(last_loan_id);
-                borrow_region.push((region2, new_loan, point));
-                reference_moves.push(new_loan);
-                debug!(
-                    "Adding generic: {:?} {:?} {:?} {}",
-                    _region1, region2, location, last_loan_id
-                );
-                last_loan_id += 1;
+                // TODO: This is used later to determine which loans are zombies. In the assignment
+                //    _1 = move _0;
+                //  assuming _0 stores a loan L0 and _1 stores the new fake loan L1, what is the zombie? I would have thought L0 becomes a zombie because the move replaced it by L1. But this line adds the newly created loan to reference_moves, which means L1 will be considered a zombie later.
+                // TODO: There are two ways to handle tuples. Consider the assignment
+                //    let _1: &'1rv mut i32;
+                //    let _2: &'2rv mut i32;
+                //    let _3: (&'3rv mut i32, &'4rv mut i32);
+                //    _3 = (move _1, move _2);
+                //  We could create a single fake loan (associated with _3) that must outlive the loans of both _1 and _2. Alternatively, (and I think this is the correct option) we could create two loans, one associated with _3.0 that must outlive the loan associated with _1, and one associated with _3.1 that must outlive the loan associated with _2.
+                let statement = mir.statement_at(location).unwrap();
+                let (lhs, _) = statement.as_assign().unwrap();
+                let lhs_places =
+                    if let Some(local) = lhs.as_local() {
+                        local.all_places(mir)
+                    } else {
+                        // TODO: The LHS may still be a tuple, even if it's not a local variable.
+                        vec![lhs.clone()]
+                    };
+                let lhs_regions = lhs_places.iter()
+                    .filter_map(|p| place_regions.for_place(p));
+                for lhs_region in lhs_regions {
+                    let loan = new_loan();
+                    reference_moves.push(loan);
+                    borrow_region.push((lhs_region, loan, point));
+                    debug!("Adding generic: _ {:?} {:?} {:?}",
+                        lhs_region, location, loan);
+                }
```

Comment:
Fake loans for assignments are created here. Right now it requires that the LHS is a local variable. For the following assignment:

    _2 = (move _3, move _4),

it lists all places below the `_2` place that have region information available -- because `_2` is a tuple, the places below `_2` are `_2.0` and `_2.1`. It then creates a fake loan for every one of these places that has a region attached.

What I don't understand is why this new fake loan is not associated with the region of the variable on the RHS that is moved into the LHS.